apps/frontend/src/components/common/SimpleTabs.tsx

```diff
@@ -58,11 +59,25 @@ const SimpleTabs = ({
 }: SimpleTabsProps) => {
   const theme = useTheme();
   const isMobile = useMediaQuery('(max-width: 500px)');
-  const [searchParams, setSearchParams] = useSearchParams();
 
-  const tab = searchParams.get('tab') ?? '0';
-  const modalTab = searchParams.get('modalTab') ?? '0';
-  const verticalTab = searchParams.get('verticalTab') ?? '0';
+  const initialParams = useMemo(
+    () => ({
+      verticalTab: null,
+      modalTab: null,
+      tab: null,
+    }),
+    []
+  );
+
+  const [typedParams, setTypedParams] = useTypeSafeSearchParams<{
+    verticalTab: number | null;
+    modalTab: number | null;
+    tab: number | null;
+  }>(initialParams);
```

Comment:
Here it's stating that the values are numbers, but how can that be the case? Internally to `useTypeSafeSearchParams` the values are pulled from the URL, so wouldn't all values be strings? And the only type conversion I can see within that hook is to turn things into strings.

Reply:
That's the idea. To make the query params consistent with types

`useTypeSafeSearchParams`
This takes care of typecasting any types to string.

Using the above hook, any component can register typed query params, which makes the application more consistent. For ex., I cannot pass proposal_id as `abcd`, since it is an integer and my implementation will throw type error. Just like how the default React useState behaves.